lib/datasource/bitbucket-tags/index.ts

```diff
@@ -0,0 +1,156 @@
+}
+
+const cacheNamespace = 'datasource-bitbucket';
+
+function getCacheKey(registryUrl: string, repo: string, type: string): string {
+  return `${getRegistryURL(registryUrl)}:${repo}:${type}`;
+}
+
+type BitbucketTag = {
+  name: string;
+  target?: {
+    date?: string;
+    hash: string;
+  };
+};
+
+// getReleases fetches list of tags for the repository
+export async function getReleases({
+  registryUrl,
+  lookupName: repo,
+}: GetReleasesConfig): Promise<ReleaseResult | null> {
+  const type = 'tags';
+  const cachedResult = await packageCache.get<ReleaseResult>(
+    cacheNamespace,
+    getCacheKey(registryUrl, repo, type)
+  );
+  // istanbul ignore if
+  if (cachedResult) {
+    return cachedResult;
+  }
+
+  const url = `/2.0/repositories/${repo}/refs/tags`;
+
+  const bitbucketTags = (
+    await bitbucketHttp.getJson<utils.PagedResult<BitbucketTag>>(url)
+  ).body;
+
+  const dependency: ReleaseResult = {
+    sourceUrl: `${ensureTrailingSlash(getRegistryURL(registryUrl))}${repo}`,
+    releases: null,
+  };
+  dependency.releases = bitbucketTags.values.map(({ name, target }) => ({
+    version: name,
+    gitRef: name,
+    releaseTimestamp: target?.date,
+  }));
+
+  const cacheMinutes = 10;
+  await packageCache.set(
+    cacheNamespace,
+    getCacheKey(registryUrl, repo, type),
+    dependency,
+    cacheMinutes
+  );
+  return dependency;
+}
+
+type BitbucketCommit = {
+  hash: string;
+  date?: string;
+};
+
+// getTagCommit fetched the commit has for specified tag
+async function getTagCommit(
+  registryUrl: string,
+  repo: string,
+  tag: string
+): Promise<string | null> {
+  const type = `tag-${tag}`;
+  const cachedResult = await packageCache.get<string>(
+    cacheNamespace,
+    getCacheKey(registryUrl, repo, type)
+  );
+  // istanbul ignore if
+  if (cachedResult) {
+    return cachedResult;
+  }
+
+  const url = `/2.0/repositories/${repo}/refs/tags/${tag}`;
+
+  const bitbucketTag = (await bitbucketHttp.getJson<BitbucketTag>(url)).body;
+
+  const hash = bitbucketTag.target.hash;
+
+  const cacheMinutes = 10;
+  await packageCache.set(
+    cacheNamespace,
+    getCacheKey(registryUrl, repo, type),
+    hash,
+    cacheMinutes
+  );
+
+  return hash;
+}
+
+// getDigest fetched the latest commit for repository main branch
+// however, if newValue is provided, then getTagCommit is called
+export async function getDigest(
+  { lookupName: repo, registryUrl }: Partial<DigestConfig>,
+  newValue?: string
+): Promise<string | null> {
+  if (newValue?.length) {
+    return getTagCommit(registryUrl, repo, newValue);
+  }
+
+  const type = 'digest';
+  const cachedResult = await packageCache.get<string>(
+    cacheNamespace,
+    getCacheKey(registryUrl, repo, type)
+  );
+  // istanbul ignore if
+  if (cachedResult) {
+    return cachedResult;
+  }
+
+  const mainBranch = (
+    await bitbucketHttp.getJson<utils.RepoInfoBody>(`/2.0/repositories/${repo}`)
+  ).body.mainbranch.name;
```

Comment:
This should be cached too

Reply:
Ok, will do. 

However, this value is already fetched in the repo config here https://github.com/renovatebot/renovate/blob/e3b7c20b260564c7e3e28906cab5393353618669/lib/platform/bitbucket/index.ts#L141

however, importing the `lib/platform/bitbucket/index.ts` introduced the dependency cycle which I didn't know how to resolve without some refactoring, and I'm not that good with TypeScript to do it.
Maybe you can give me some suggestions?

Reply:
do not reuse that value, it can be from other repo. you need to fetch as now, but ideally cache the response and reuse it.